modules/percolator/src/main/java/org/elasticsearch/percolator/PercolatorFieldMapper.java

```diff
@@ -405,4 +482,26 @@ static void verifyQuery(QueryBuilder queryBuilder) {
         }
     }
 
+    static byte[] encodeRange(String rangeFieldName, byte[] minEncoded, byte[] maxEncoded) {
+        assert minEncoded.length == maxEncoded.length;
+
+        byte[] bytes = new byte[BinaryRange.BYTES * 2];
+        int offset = BinaryRange.BYTES - minEncoded.length;
+        if (offset >= 8) {
```

Comment:
Actually I was thinking of doing something like this so that we use as many bytes of the hash as possible. For instance in the case of ints/floats, we'd still put 12 bytes of the hash in the values. And if we ever come up with types that need 12 bytes, then we'd automatically put 4 bytes of the hash in values.
```
byte[] bytes = new byte[BinaryRange.BYTES * 2];
BytesRef fieldAsBytesRef = new BytesRef(rangeFieldName);
MurmurHash3.Hash128 hash = new MurmurHash3.Hash128();
MurmurHash3.hash128(fieldAsBytesRef.bytes, fieldAsBytesRef.offset, fieldAsBytesRef.length, 0, hash);
ByteBuffer bb = ByteBuffer.wrap(bytes);
bb.put(hash.h1).put(hash.h2).put(hash.h1).put(hash.h2);
assert bb.position() == bb.limit();
int offset = BinaryRange.BYTES - minEncoded.length;
System.arraycopy(minEncoded, 0, bytes, offset, minEncoded.length);
System.arraycopy(maxEncoded, 0, bytes, BinaryRange.BYTES + offset, minEncoded.length);
```